backend/schemamirror/service.go

```diff
@@ -0,0 +1,229 @@
+	}
+	for _, realm := range s.eventSource.CanonicalView().Realms {
+		for _, m := range realm.Modules {
+			d := m.GetRuntime().GetDeployment()
+			if d == nil {
+				continue
+			}
+			if deploymentKey == d.DeploymentKey {
+				return connect.NewResponse(&ftlv1.GetDeploymentResponse{
+					Schema: m.ToProto(),
+				}), nil
+			}
+		}
+	}
+	for _, cs := range s.eventSource.ActiveChangesets() {
+		for _, m := range cs.InternalRealm().Modules {
+			d := m.GetRuntime().GetDeployment()
+			if d == nil {
+				continue
+			}
+			if deploymentKey == d.DeploymentKey {
+				return connect.NewResponse(&ftlv1.GetDeploymentResponse{
+					Schema: m.ToProto(),
+				}), nil
+			}
+		}
+	}
+	return nil, errors.Wrapf(err, "failed to find deployment %s", deploymentKey)
+}
+
+func (s *Service) UpdateDeploymentRuntime(context.Context, *connect.Request[ftlv1.UpdateDeploymentRuntimeRequest]) (*connect.Response[ftlv1.UpdateDeploymentRuntimeResponse], error) {
+	return nil, connect.NewError(connect.CodeUnimplemented, errors.New("schema mirror is read only"))
+}
+
+func (s *Service) CreateChangeset(context.Context, *connect.Request[ftlv1.CreateChangesetRequest]) (*connect.Response[ftlv1.CreateChangesetResponse], error) {
+	return nil, connect.NewError(connect.CodeUnimplemented, errors.New("schema mirror is read only"))
+}
+
+func (s *Service) PrepareChangeset(context.Context, *connect.Request[ftlv1.PrepareChangesetRequest]) (*connect.Response[ftlv1.PrepareChangesetResponse], error) {
+	return nil, connect.NewError(connect.CodeUnimplemented, errors.New("schema mirror is read only"))
+}
+
+func (s *Service) CommitChangeset(context.Context, *connect.Request[ftlv1.CommitChangesetRequest]) (*connect.Response[ftlv1.CommitChangesetResponse], error) {
+	return nil, connect.NewError(connect.CodeUnimplemented, errors.New("schema mirror is read only"))
+}
+
+func (s *Service) DrainChangeset(context.Context, *connect.Request[ftlv1.DrainChangesetRequest]) (*connect.Response[ftlv1.DrainChangesetResponse], error) {
+	return nil, connect.NewError(connect.CodeUnimplemented, errors.New("schema mirror is read only"))
+}
+
+func (s *Service) FinalizeChangeset(context.Context, *connect.Request[ftlv1.FinalizeChangesetRequest]) (*connect.Response[ftlv1.FinalizeChangesetResponse], error) {
+	return nil, connect.NewError(connect.CodeUnimplemented, errors.New("schema mirror is read only"))
+}
+
+func (s *Service) RollbackChangeset(context.Context, *connect.Request[ftlv1.RollbackChangesetRequest]) (*connect.Response[ftlv1.RollbackChangesetResponse], error) {
+	return nil, connect.NewError(connect.CodeUnimplemented, errors.New("schema mirror is read only"))
+}
+
+func (s *Service) FailChangeset(context.Context, *connect.Request[ftlv1.FailChangesetRequest]) (*connect.Response[ftlv1.FailChangesetResponse], error) {
+	return nil, connect.NewError(connect.CodeUnimplemented, errors.New("schema mirror is read only"))
+}
```

Comment:
I wonder if at some point we should pull these out into a separate service - so we'd have a read-only schema service and a mutable one.

Reply:
That'd be nicer